pyproject.toml

```diff
@@ -184,6 +184,80 @@ exclude = '''
 )
 '''
 
+[tool.ruff]
+line-length = 88
+
+select = [
+  # pyflakes
+  "F",
+  # pycodestyle
+  "E",
+  "W",
+  # flake8-bugbear
+  "B",
+]
+
+ignore = [
+  # space before : (needed for how black formats slicing)
+  # "E203",  # not yet implemented
+  # module level import not at top of file
+  "E402",
+  # do not assign a lambda expression, use a def
+  "E731",
+  # line break before binary operator
+  # "W503",  # not yet implemented
+  # line break after binary operator
+  # "W504",  # not yet implemented
+  # controversial
+  "B006",
+  # controversial
+  "B007",
+  # controversial
+  "B008",
+  # setattr is used to side-step mypy
+  "B009",
+  # getattr is used to side-step mypy
+  "B010",
+  # tests use assert False
+  "B011",
+  # tests use comparisons but not their returned value
+  "B015",
+  # false positives
+  "B019",
+  # Use of functools.lru_cache or functools.cache on methods can lead to memory leaks.
+  "B020",
+  # Loop control variable overrides iterable it iterates
+  "B023",
+  # Functions defined inside a loop must not use variables redefined in the loop
+  # "B301",  # not yet implemented
+
+  # Additional checks that don't pass yet
+  # `zip()` without an explicit `strict=` parameter
+  "B905",
+  # Within an except clause, raise exceptions with ...
+  "B904",
+]
+
+exclude = [
+  "*.pyi",
+  "doc/sphinxext/*.py",
+  "doc/build/*.py",
+  "doc/temp/*.py",
+  ".eggs/*.py",
+  "versioneer.py",
+  # exclude asv benchmark environments from linting
+  "env",
+]
+
+[tool.ruff.per-file-ignores]
+# ignoring multiline is not yet supported (https://github.com/charliermarsh/ruff/issues/1052)
+"pandas/util/__init__.py" = ["F401"]
+"pandas/tests/extension/base/__init__.py" = ["F401"]
+# undefined name with global (https://github.com/charliermarsh/ruff/issues/960)
+"pandas/io/clipboard/__init__.py" = ["F821"]
```

Comment:
Should also be fixed since 0.0.171

Reply:
yup, thanks! all the per-file-ignores ones could be removed in fact, the Literal one was just due to `Literal` having been imported from `pandas._typing` rather than `typing`, so a static tool doesn't recognise it

Reply:
Cool, thanks!

Reply:
Looking at `pandas/core/reshape/merge.py`, this may no longer be relevant, but as a heads up, we do now support a setting specifically for this use-case, to mark modules as "`typing` redefinitions`.

Issue that motivated it (from Airflow): https://github.com/charliermarsh/ruff/issues/1744
Docs: https://github.com/charliermarsh/ruff#typing-modules